hs_core/tests/api/native/test_reorder_authors_management_command.py

```diff
@@ -0,0 +1,207 @@
+        resource.create_metadata_element(self.res.short_id, 'creator', name='Lisa McWill')
+        resource.create_metadata_element(self.res.short_id, 'creator', name='Kelly Anderson')
+        resource.create_metadata_element(self.res.short_id, 'creator', name='Mark Miller')
+
+        self.update_command = "reorder_authors"
+
+    def tearDown(self):
+        super(TestReorderAuthorsCommand, self).tearDown()
+        User.objects.all().delete()
+        Group.objects.all().delete()
+        GenericResource.objects.all().delete()
+        Creator.objects.all().delete()
+
+    def test_command_doesnt_alter_if_aready_correct(self):
+        """
+        Testing author order is maintained if they were already correct
+        """
+        citation_original = self.res.get_citation()
+
+        self.assertEqual(self.res.get_citation(), citation_original)
+        for index, creator in enumerate(self.res.metadata.creators.all(), start=1):
+            self.assertEqual(index, creator.order)
+
+        # run  update command to fix author order
+        call_command(self.update_command)
+
+        self.assertEqual(self.res.get_citation(), citation_original)
+        for index, creator in enumerate(self.res.metadata.creators.all(), start=1):
+            self.assertEqual(index, creator.order)
+
+    def test_command_alters_if_incorrect(self):
+        """
+        Testing author order is updated if it was incorrect
+        """
+
+        # Intentionally make a creator list with duplicate orders
+        john = self.res.metadata.creators.filter(name="John Smith").first()
+        lisa = self.res.metadata.creators.filter(name="Lisa McWill").first()
+        self.assertEqual(john.order, 2)
+        john.order = 3
+        john.save()
+        self.assertEqual(john.order, lisa.order)
+
+        # run  update command to fix author order
+        call_command(self.update_command)
+
+        for index, creator in enumerate(self.res.metadata.creators.all(), start=1):
+            self.assertEqual(index, creator.order)
+
+    def test_command_alters_missing_first_author(self):
+        """
+        Testing author order is updated if there was a missing first author
+        """
+
+        # Intentionally make a creator list with duplicate orders
+        john = self.res.metadata.creators.filter(name="John Smith").first()
+        first_author = self.res.metadata.creators.filter(order=1).first()
+        self.assertEqual(first_author.order, 1)
+        first_author.order = 2
+        first_author.save()
+        self.assertEqual(john.order, first_author.order)
+        self.assertEqual(first_author.name, "Creator_LastName, Creator_FirstName")
+
+        # run  update command to fix author order
+        call_command(self.update_command)
+
+        for index, creator in enumerate(self.res.metadata.creators.all(), start=1):
+            self.assertEqual(index, creator.order)
+
+        self.assertEqual(first_author.name, "Creator_LastName, Creator_FirstName")
```

Comment:
After running the command, we have to fetch the `first_author` again from db (line 93) to check the author name.

Reply:
Indeed, thanks